examples/3_write_serial.cpp

```diff
@@ -49,7 +49,7 @@ int main(int argc, char *argv[])
     cout << "Created a scalar mesh Record with all required openPMD "
             "attributes\n";
 
-    Datatype datatype = determineDatatype(shareRaw(global_data));
+    Datatype datatype = determineDatatype(global_data.data());
```

Comment:
determineDatatype now also accepts (normal) pointers as input.

Reply:
Since we are `shared_ptr` geeks in the public API, I would also add a `shared_ptr` overload to this.

Reply:
Let's also add a new `T_ContiguousContainer` overload for this one, for symmetry with `storeChunk`?

Reply:
Maybe we can keep it simple by documenting to use:
```C++
determineDatatype<decltype(global_data)::value_type>()
```